src/App.tsx

```diff
@@ -1,223 +1,222 @@
-              />
-            </label>
-
-            <span data-cy="TodoTitle" className="todo__title">HTML</span>
-            <button
-              type="button"
-              className="todo__remove"
-              data-cy="TodoDeleteButton"
-            >
-              ×
-            </button>
+    await deleteTodo(todoId);
 
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
+    await loadTodos();
 
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
+    setLoadingTodoIds([]);
+  };
 
-            <span data-cy="TodoTitle" className="todo__title">CSS</span>
+  const deleteComplitedTodos = async () => {
+    setLoadingTodoIds(prevIds => [
+      ...prevIds,
+      ...completedTodos.map(todo => todo.id)]);
 
-            <button
-              type="button"
-              className="todo__remove"
-              data-cy="TodoDeleteButton"
-            >
-              ×
-            </button>
+    setError(Error.None);
+
+    try {
+      await Promise.all(completedTodos.map(todo => (
+        deleteTodo(todo.id)
+      )));
+
+      await loadTodos();
+    } catch {
+      setError(Error.Delete);
+    }
+
+    setLoadingTodoIds([]);
+  };
 
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
-
-            <form>
-              <input
-                data-cy="TodoTitleField"
-                type="text"
-                className="todo__title-field"
-                placeholder="Empty todo will be deleted"
-                defaultValue="JS"
-              />
-            </form>
-
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
-
-            <span data-cy="TodoTitle" className="todo__title">React</span>
+  const updateTodoStatus = async (updatedTodo: Todo) => {
```

Comment:
when i try to mark todo as completed it just deletes it

Reply:
maybe you had turned on filter on 'active', cuz i tried and it is working ok now)